bundler/lib/bundler/plugin.rb

```diff
@@ -100,23 +102,34 @@ def list
     #
     # @param [Pathname] gemfile path
     # @param [Proc] block that can be evaluated for (inline) Gemfile
-    def gemfile_install(gemfile = nil, &inline)
+    def gemfile_install(gemfile = nil, unlock: false, &inline)
+      was_deployment = Bundler.settings[:deployment]
       Bundler.settings.temporary(frozen: false, deployment: false) do
         builder = DSL.new
         if block_given?
           builder.instance_eval(&inline)
         else
           builder.eval_gemfile(gemfile)
         end
-        builder.check_primary_source_safety
-        definition = builder.to_definition(nil, true)
 
-        return if definition.dependencies.empty?
+        return if builder.dependencies.empty?
 
-        plugins = definition.dependencies.map(&:name).reject {|p| index.installed? p }
-        installed_specs = Installer.new.install_definition(definition)
+        Bundler.configure_gem_home_and_path
```

Comment:
Is this necessary?

Reply:
No longer applicable with the new approach.